src/app/features/course/courses-section/courses-section.component.ts

```diff
@@ -1,16 +1,199 @@
-import {Component, OnInit} from '@angular/core';
+import {
+  Component,
+  OnChanges,
+  OnInit,
+  DoCheck,
+  AfterContentInit,
+  AfterContentChecked,
+  AfterViewInit,
+  AfterViewChecked,
+  OnDestroy
+} from '@angular/core';
+import {Course} from '../../../core/models/course';
 
 @Component({
   selector: 'app-section',
   templateUrl: './courses-section.component.html',
   styleUrls: ['./courses-section.component.scss']
 })
-export class CoursesSectionComponent implements OnInit {
+export class CoursesSectionComponent implements OnChanges,
+                                                OnInit,
+                                                DoCheck,
+                                                AfterContentInit,
+                                                AfterContentChecked,
+                                                AfterViewInit,
+                                                AfterViewChecked,
+                                                OnDestroy {
+  coursesToDisplay: Array<Course> = [];
```

Comment:
that's ok, but more popular writing is coursesToDisplay: Course[] = [];

Reply:
done, thanks!